Documenting new algorithm Detect Dataset Changes

rosaguilar opened the pull request:
Goal:

Ticket(s): Fix #4788

- [ ] Backport to LTR documentation is required

m-kuhn commented:
Wow, nice work. I love the image to explain the alg.

It looks like the features tagged "deleted" are still there in set 2. Is this an error in my understanding, the alg or the image?

rosaguilar commented:
> Wow, nice work. I love the image to explain the alg.
> 
> It looks like the features tagged "deleted" are still there in set 2. Is this an error in my understanding, the alg or the image?

Thanks for catching this. 

> Wow, nice work. I love the image to explain the alg.
> 
> It looks like the features tagged "deleted" are still there in set 2. Is this an error in my understanding, the alg or the image?

Thanks for catching this. A new image was uploaded.

rosaguilar commented:
> Looks fine to me.

@havatv, Can you please check (in the code) the names of "unchanged", "deleted", and "added".
I assumed those names but I was waiting for your feedback. I already sent a mail to Nyall but not reply yet.

havatv commented:
> @havatv, Can you please check (in the code) the names of "unchanged", "deleted", and "added".
> I assumed those names but I was waiting for your feedback. I already sent a mail to Nyall but not reply yet.

Both the labels and names are correct.
However, the three output parameters that give the count for each group should also be included:

* "UNCHAGED_COUNT" ( "Count of unchanged features" )
* "ADDED_COUNT" ( "Count of features added in revised layer" )
* "DELETED_COUNT" ("Count of features deleted from original layer")

 (@nyalldawson, note the spelling mistake in the code: "UNCHAGED" -> "UNCHANGED"):

nyalldawson commented:
>  I already sent a mail to Nyall but not reply yet.

Sorry, don't recall receiving that email

>  note the spelling mistake in the code: "UNCHAGED" -> "UNCHANGED"

oops will fix, thanks!

nyalldawson commented:
@rosaguilar 
> Can you please check (in the code) the names of "unchanged", "deleted", and "added".
I assumed those names but I was waiting for your feedback. I already sent a mail to Nyall but not reply yet.

sorry, found that email. It's a QGIS bug, I'm fixing now

rosaguilar commented:
Thanks @havatv and @nyalldawson. I added the missing output parameters (log), not sure of the labels. I also replaced the figure to delete the caption (I think it is not needed).

havatv commented on docs/user_manual/processing_algs/qgis/vectorgeneral.rst:
```diff
@@ -537,7 +537,115 @@ Outputs
+       * Save to PostGIS Table...
+
+       The file encoding can also be changed here.
+   * - **Unchanged count**
```
```suggestion
   * - **Count of unchanged features**
```

havatv commented on docs/user_manual/processing_algs/qgis/vectorgeneral.rst:
```diff
@@ -537,7 +537,115 @@ Outputs
+     - ``UNCHANGED_COUNT``
+     - [number]
+     - Count of unchanged features.
+   * - **Added count**
```
```suggestion
   * - **Count of features added in revised layer**
```

havatv commented on docs/user_manual/processing_algs/qgis/vectorgeneral.rst:
```diff
@@ -537,7 +537,115 @@ Outputs
+     - ``ADDED_COUNT``
+     - [number]
+     - Count of features added in revised layer.
+   * - **Deleted count**
```
```suggestion
   * - **Count of features deleted from original layer**
```

havatv commented:
> Thanks @havatv and @nyalldawson. I added the missing output parameters (log), not sure of the labels. I also replaced the figure to delete the caption (I think it is not needed).

It was actually the lables that were placed in parenthesis after the names. See my suggestions.

rosaguilar commented:
> > Thanks @havatv and @nyalldawson. I added the missing output parameters (log), not sure of the labels. I also replaced the figure to delete the caption (I think it is not needed).
> 
> It was actually the lables that were placed in parenthesis after the names. See my suggestions.

Thanks, suggestions applied.

rosaguilar commented:
> > Thanks @havatv and @nyalldawson. I added the missing output parameters (log), not sure of the labels. I also replaced the figure to delete the caption (I think it is not needed).
> 
> It was actually the lables that were placed in parenthesis after the names. See my suggestions.

Sorry. I thought that it was the description :)

havatv commented:
> Sorry. I thought that it was the description :)

No problem! Thanks for your efforts.

DelazJ commented on docs/user_manual/processing_algs/qgis/vectorgeneral.rst:
```diff
@@ -537,7 +537,115 @@ Outputs
+     - Name
+     - Type
+     - Description
+   * - **Unchanged features**
```
@havatv Are these outputs shown in the dialog no longer part of the parameters (eg https://docs.qgis.org/3.10/en/docs/user_manual/processing_algs/qgis/vectorcreation.html#import-geotagged-photos)?

havatv replied:
@rosaguilar, the three output datasets/layers must also be described among the parameters (they should be present both in the Parameters and Output tables) - see for instance @DelazJ's reference for how it is normally done.

rosaguilar replied:
Thanks, @DelazJ, and @havatv.

havatv replied:
The *Type* of the output should be [same as input] (not [vector: any]) in both the Parameters and Output tables.

rosaguilar replied:
ok. but what is **input**?
I followed the example (https://docs.qgis.org/3.10/en/docs/user_manual/processing_algs/qgis/vectorcreation.html#import-geotagged-photos).  or 
https://docs.qgis.org/3.10/en/docs/user_manual/processing_algs/qgis/vectorcreation.html#id34
I was thinking of [same as Original], as "Original" is the name of the "Original layer".
Its confusing

m-kuhn replied:
I've left a suggestion the way I think it should be

rosaguilar replied:
Thanks @m-kuhn. Suggestion added.

m-kuhn commented on docs/user_manual/processing_algs/qgis/vectorgeneral.rst:
```diff
@@ -537,7 +537,127 @@ Outputs
+       * 1 --- Tolerant Match (Topological Equality): geometries are considered equal
+   * - **Unchanged features**
+     - ``UNCHANGED``
+     - [vector: any]
```
```suggestion
     - [vector: same as Original layer]
```

rosaguilar commented on docs/user_manual/processing_algs/qgis/vectorgeneral.rst:
```diff
@@ -537,7 +537,127 @@ Outputs
+       The file encoding can also be changed here.
+   * - **Deleted features**
+     - ``DELETED``
+     - [vector: any]
```
Thanks @m-kuhn,  Same here.

rosaguilar commented on docs/user_manual/processing_algs/qgis/vectorgeneral.rst:
```diff
@@ -537,7 +537,127 @@ Outputs
+       The file encoding can also be changed here.
+   * - **Added features**
+     - ``ADDED``
+     - [vector: any]
```
Thanks @m-kuhn,  this was already done in my previous commit.

m-kuhn replied:
I don't think so. Click on the "Commit suggestion" button -- or even better use "Add suggestion to batch" on the files tab to create a new commit with the 3 remaining suggestions.

rosaguilar replied:
> I don't think so. Click on the "Commit suggestion" button -- or even better use "Add suggestion to batch" on the files tab to create a new commit with the 3 remaining suggestions.

@m-kuhn, sorry for the questions, I am trying to understand how this works. Commiting suggestions is te only way to address changes? I edit directly the file, after your first suggestion, and I changed the three ocurrences

m-kuhn replied:
You can also modify it yourself. But you need to change 6 occurrences (3 x parameters, 3 x outputs)

rosaguilar replied:
@m-kuhn, thanks. Hopefully, it is done now.

m-kuhn commented:
Awesome work @rosaguilar :rocket:

rosaguilar commented:
> Awesome work @rosaguilar 🚀

Thanks, @m-kuhn for reviewing and for the encouraging words.

m-kuhn commented:
@havatv @DelazJ is there anything missing in here from your perspective?
For me, it's good to ship :ship:

DelazJ commented:
Thanks @rosaguilar Great work as usual.

rosaguilar commented:
Thanks to all the reviewers too.
